Source/WebCore/accessibility/AccessibilityARIAGridRow.cpp

```diff
@@ -71,7 +71,7 @@ AXCoreObject::AccessibilityChildrenVector AccessibilityARIAGridRow::disclosedRow
     // The contiguous disclosed rows will be the rows in the table that 
     // have an aria-level of plus 1 from this row.
     RefPtr parent = parentObjectUnignored();
-    if (auto* axTable = dynamicDowncast<AccessibilityTable>(*parent); !axTable || !axTable->isExposable())
+    if (SUPPRESS_UNCOUNTED_LOCAL auto* axTable = dynamicDowncast<AccessibilityTable>(*parent); !axTable || !axTable->isExposable())
```

Comment:
Are you using an outdated version of llvm? This should not be reported as a warning by static analysis since `isExposable` is trivial and inline. We should also not be adding `SUPPRESS_UNCOUNTED_LOCAL` for false positives.

Since this PR adds a lot of `SUPPRESS_UNCOUNTED_LOCAL`, it is likely not correct. My bet is that you're using a llvm version which doesn't have our latest false positive fixes. If not, we should look into fixing those false positives in llvm (cc @rniwa).

Reply:
Note that Fred is new to smart pointer work :) Hmm, I forgot to tell him about not using SUPPRESS_UNCOUNTED_LOCAL, sorry.

Our setup is to compile latest llvm project on Linux, then run the built clang-tidy on a local WPE Debug build.

A colleague tried a similar approach on OS X but only got results for the JSC part, not WebKit.

If you have pointers how to make things work on OS X, I'd be very interested!

Reply:
If you're getting a warning for this variable, it's probably a bug in WebKit checkers.

Reply:
@cdumez @rniwa: Thanks for the feedback and sorry about that! I guess I was confused by some previous PRs that do add this kind of SUPPRESS_UNCOUNTED_LOCAL and was also not aware that our own Linux setup gives different result. I have removed all the SUPPRESS_UNCOUNTED_LOCAL things, so it only remains the ones that clearly use non-trivial functions.

That said, I still see several usages of [SUPPRESS_UNCOUNTED_LOCAL](https://searchfox.org/wubkat/search?q=SUPPRESS_UNCOUNTED_LOCAL&path=&case=false&regexp=false) so I'd really like to understand why they are necessary (if they still are!). Really, I believe we should add a comment with an explanation when we add such usages. Anyway checking the existing ones:

* ExceptionScope.h: Added in https://github.com/WebKit/WebKit/pull/28111 ; I guess this intentional exception for the JSC code?

* ValidatedFormListedElement::updateValidity(): Added in https://github.com/WebKit/WebKit/pull/24676 ; I believe all implementations of fasHTMLElement() are trivial inline functions, is that something difficult to detect by the static analyzer?

* IntersectionObserver::isReachableFromOpaqueRoots(): Discussed on https://github.com/WebKit/WebKit/pull/27926#pullrequestreview-2031561732 ; it seems the static analyzer cannot detect trivial functions that involve non-inline functions?

* LocalFrame.cpp: Added in https://github.com/WebKit/WebKit/pull/27870 ; all of them only seem to be trivial functions. Maybe for the last one where `RenderStyle::usedUserSelect()` is non-inline and so is hard to detect by the static analyzer?

* SVGElement.cpp: Done in https://github.com/WebKit/WebKit/pull/24460 and https://github.com/WebKit/WebKit/pull/24620 ; These only seem to involve trivial inline functions, so not sure why the static analyzer cannot detect that?

* SVGLocatable.cpp: Done in https://github.com/WebKit/WebKit/pull/24460 and https://github.com/WebKit/WebKit/pull/24620 ; we don't call member of `farthest` or pass it to any function, so not sure why static analyzer would warn here. I guess the initial commit was actually meant to deal with variables svgElement / current but it seems only trivial functions are used for them (our current Linux setup generates warnings, so perhaps it was a limitation of the previous versions of the analyzer)

* SVGImageCache::imageSizeForRenderer: Done in https://github.com/WebKit/WebKit/pull/24460 and https://github.com/WebKit/WebKit/pull/24620 ; SVGImageForContainer::size seems trivial but it is non-inline so perhaps the static analyzer cannot detect that?

In general it seems there is a pattern of having functions that look trivial but could be complex enough to deceive the static analyzer. In my previous version of this PR, I was adding SUPPRESS_UNCOUNTED_LOCAL for things that involve the following functions and I wonder whether current version of the analyzer actually handle them as trivial functions:
  - AXCoreObject::isGroup()
  - AccessibilityRenderObject::node()
  - AccessibilityRole::roleValue()/AccessibilityTable::roleValue()
  - AccessibilityTable::hasGridAriaRole()
  - HashMap::ensure()

Reply:
> That said, I still see several usages of [SUPPRESS_UNCOUNTED_LOCAL](https://searchfox.org/wubkat/search?q=SUPPRESS_UNCOUNTED_LOCAL&path=&case=false&regexp=false) so I'd really like to understand why they are necessary (if they still are!).

FWIW I just tested each of the `SUPPRESS_UNCOUNTED_LOCAL` uses in `Source/WebCore` using a clang build at [13008aa45d406a65ee7adfc7672a038e4def1ad3](https://github.com/llvm/llvm-project/commit/13008aa45d406a65ee7adfc7672a038e4def1ad3) on macOS, and removing any of them generates a new `UncountedLocalVarsChecker` warning.

Reply:
To answer more specifically the original question, our recent runs on Linux and macOS (using a clang version that includes https://github.com/llvm/llvm-project/commits/13008aa45d406a65ee7adfc7672a038e4def1ad3/?author=rniwa) still give warnings for `axTable`:

```
Source/WebCore/accessibility/AccessibilityARIAGridRow.cpp:74:15: warning: Local variable 'axTable' is uncounted and unsafe [clang-analyzer-alpha.webkit.UncountedLocalVarsChecker]
Source/WebCore/accessibility/AccessibilityARIAGridRow.cpp:103:15: warning: Local variable 'axTable' is uncounted and unsafe [clang-analyzer-alpha.webkit.UncountedLocalVarsChecker]
```